jerry-core/ecma/base/ecma-globals.h

```diff
@@ -1121,6 +1133,22 @@ typedef struct
 
 #endif /* !CONFIG_ECMA_LCACHE_DISABLE */
 
+#ifndef CONFIG_DISABLE_TYPEDARRAY_BUILTIN
+
+/**
+ * Internal properties of TypedArray Object
+ */
+typedef struct
+{
+  /** The arraybuffer object behind the typedarray */
+  ecma_object_t *viewed_arraybuffer_p;
```

Comment:
Ah, so a data view can be a sub-array. Interesting. Then what about this layout:

Use the normal extended object when offset = 0, and length = buffer.length. I think this is the most common case, and we can save space.

When this is not true, allocate an extra 8 byte, where the offset and length are stored.